code/go/pkg/specerrors/errors.go

```diff
@@ -0,0 +1,78 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package specerrors
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ValidationError is the interface that every validation error must implement.
+type ValidationError interface {
+	error
+
+	// Code returns a unique identifier of this error.
+	Code() string
+}
+
+// ValidationPathError is the interface that validation errors related to paths must implement.
+type ValidationPathError interface { // TODO no validation error using this interface yet
+	// File returns the file path where the error was raised.
+	File() string
+}
+
+// ValidationSeverityError is the interface that validation errors related to severities must implement.
+type ValidationSeverityError interface { // TODO no validation error using this interface yet
+	// File returns the file path where the error was raised.
+	Severity() string
+}
+
+// ValidationErrors is an error that contains an iterable collection of validation error messages.
+type ValidationErrors []ValidationError
+
+// Collect filters the validation errors using the function given as a parameter.
+func (ve ValidationErrors) Collect(collect func(elem ValidationError) bool) (ValidationErrors, ValidationErrors) {
```

Comment:
Nit. Use names here to document what is each ValidationErrors, or use a struct similar to ProcessResult.
```suggestion
func (ve ValidationErrors) Collect(collect func(elem ValidationError) bool) (collected ValidationErrors, excluded ValidationErrors) {
```